### PR TITLE
Solved "Add cogwheel icon for (future) user settings #121"

### PR DESCRIPTION
**Pull request by vandrongelen**

### CODE REVIEW

**Comment by femans** [comment]
## TL;DR: 
- stuff is implemented but needs polishing up/debugging before merging. 
- Run [nsfw-server](https://github.com/ipfs-search/nsfw-server) in a separate terminal to make this explicit image blurring work. Should be easy.

### long version:
@vandrongelen I added functionality to the buttons. The nsfw filter is still based on having a separate node server running that does the work for this, which can be found here: [nsfw-server](https://github.com/ipfs-search/nsfw-server). If you fire up this project in a separate terminal, it should work.

Before we can merge this, a number of things need to be made to work:

- adding the menu to the detail view. (I tried to do this but for some reason it does not show up). Maybe you @vandrongelen can look at this
- making sure the dark theme is behaving correctly now; also good for you to look at
For me:
- debugging of the functionality; it still has some issues, for example when changing the search query
- making the NSFW server location configurable in the build; ATM it is only pointing at localhost:3000, this can not be deployed.

For further progress, we should also decide on a threshold value for the nsfw filter. Right now I put it on 30%, which seems to be adequate. It is a bit of a hassle to see what are the values for a given image, maybe we could include it in the metadata panel on the detail page (still to be implemented) or in a tooltip. 

I am still concerned about the server architecture and feel it would be better to have the NSFW classifier running locally, to increase the speed. We should test this out in the field, when we have caching enabled for the nsfw server.

**Comment by dokterbob** [comment]
Outstanding issues
- [ ] 2 CodeClimate smells
- [ ] Cogwheel on homepage not visible (wrong colour)
- [ ] Darkmode should default to detected current browser settings, not to either dark or light

**Comment by femans** [comment]
@dokterbob This is stable, cleaned and merged now.

- added a message to blurred images in the list view. 
- added the settings cog wheel in the detail view.
- fixed the color of the cogwheel in home view
- fixed the default for the dark mode. Notice that there is a hiatus in the UX, because there should be a third option for darkmode: system setting. I could not make a simple fix for this.
- Refactored the blur mixin into a composition module

File: src/helpers/typeHelper.js [hunk 1]
```diff
@@ -1,9 +1,3 @@
-import DocumentDetail from '@/components/results/detail/DocumentDetail';
```
Comment (femans) [inline src/helpers/typeHelper.js:1]: This caused an import cycle, so I moved it to the detail view, this is the only usage of it.
